pkg/adapter/mtping/adapter.go

```diff
@@ -91,6 +92,20 @@ func GetNoShutDownAfterValue() int {
 	return 55 // seems a reasonable default
 }
 
+func GetSinkWaitDuration() int {
+	str := os.Getenv(EnvSinkWaitDuration)
+	duration := -1 // internally forever
```

Comment:
What about removing this variable and just returning the value inside the if and outside it?

Reply:
Generally functions should have one entry one exit except maybe some quick parameter checks but I'll change

Reply:
Go does not follow this pattern. A good read: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88